tests/jballerina-unit-test/src/test/resources/test-src/parser/listener-declaration-without-type-negative.bal

```diff
@@ -0,0 +1,7 @@
+import ballerina/http;
+
+listener lstnr = new http:MockListener(8080);
+public listener lstnr2 = new http:MockListener(8081);
+listener http:Listener lstnr3 = new http:MockListener(8082);
+
+private listener lstnr4 = new http:Listerner(8083)
```

Comment:
Having `private` modifier results a syntax error. Because of that, the actual semantic error is hidden. Please update the test case.

Also include test cases like,

listener x = new T ();  // Here T is not defined. 
listener y = new P(); // Here is P is defined, but not a subtype of listener interface.